packages/react-reconciler/src/ReactFiberHooks.new.js

```diff
@@ -1706,6 +1708,43 @@ function rerenderOpaqueIdentifier(): OpaqueIDType | void {
   return id;
 }
 
+function mountRefresh() {
+  const cache: Cache | null = readContext(CacheContext);
+  return mountCallback(refreshCache.bind(null, cache), [cache]);
+}
+
+function updateRefresh() {
+  const cache: Cache | null = readContext(CacheContext);
+  return updateCallback(refreshCache.bind(null, cache), [cache]);
+}
+
+function refreshCache(cache: Cache | null) {
```

Comment:
We're gonna need a way to seed the cache from a mutation response. E.g. `refresh(fnKey, valueFromResponse)`.

Reply:
This is something DevTools would make use of as well (if we decided to test this mechanism in DeeTools).

Reply:
Added in new commit